src/ray/raylet/scheduling_policy.cc

```diff
@@ -53,7 +53,25 @@ std::unordered_map<TaskID, ClientID> SchedulingPolicy::Schedule(
     const auto &resource_demand = spec.GetRequiredPlacementResources();
     const TaskID &task_id = spec.TaskId();
 
-    // TODO(atumanov): try to place tasks locally first.
+    // Try to place tasks locally first.
+    const auto &local_resources = cluster_resources[local_client_id];
+    ResourceSet available_local_resources =
+        ResourceSet(local_resources.GetAvailableResources());
+    // We have to subtract the current "load" because we set the current "load"
+    // to be the resources used by tasks that are in the
+    // `SchedulingQueue::ready_queue_` in NodeManager::HandleWorkerAvailable's
+    // call to SchedulingQueue::GetResourceLoad.
+    available_local_resources.SubtractResources(local_resources.GetLoadResources());
+    if (resource_demand.IsSubset(available_local_resources)) {
+      // This node is a feasible candidate.
+      decision[task_id] = local_client_id;
+
+      ResourceSet new_load(cluster_resources[local_client_id].GetLoadResources());
+      new_load.AddResources(resource_demand);
+      cluster_resources[local_client_id].SetLoadResources(std::move(new_load));
```

Comment:
Can we mutate the local resources instead of moving a new one in?

Reply:
Hmm I was just copying what was here before. From the interface, it doesn't look like there is a way to do that.